include/greentea-client/test_env.h

```diff
@@ -80,24 +80,144 @@ extern const char *GREENTEA_TEST_ENV_LCOV_START;
 /**
  *  Greentea-client related API for communication with host side
  */
+
+/**
+ * Handshake with the host and send setup data (timeout and host test name). Allows you to preserve sync UUID.
+ *
+ * @details This function will send a preamble to the host.
+ *
+ * @note After the host test name is received the host will invoke host test script
+ *       and add host test's callback handlers to main event loop.
+ * @note This function is blocking.
+ */
 void GREENTEA_SETUP_UUID(const int timeout, const char *host_test_name, char *buffer, size_t size);
-void GREENTEA_TESTSUITE_RESULT(const int);
+
+/**
+ * Notify the host (__exit message) side that test suite execution was complete.
+ *
+ * @note If __exit is not received by the host side we will assume TIMEOUT.
+ *
+ * @param result Test suite result
+ */
+void GREENTEA_TESTSUITE_RESULT(const int result);
+
+/**
+ *  Test Case support
+ */
+
+/**
+ * Notify the host side that test case started.
+ *
+ * @param test_case_name Test case name
+ */
 void GREENTEA_TESTCASE_START(const char *test_case_name);
-void GREENTEA_TESTCASE_FINISH(const char *test_case_name, const size_t passes, const size_t failed);
+
+/**
+ * Notify host side that test case finished.
+ *
+ * @param test_case_name Test case name
+ * @param passes Number of test passes
+ * @param failures Number of test failures
+ */
+void GREENTEA_TESTCASE_FINISH(const char *test_case_name, const size_t passes, const size_t failures);
 
 /**
  *  Test suite result related notification API
  */
-void greentea_send_kv(const char *, const int);
-void greentea_send_kv(const char *, const int, const int);
-void greentea_send_kv(const char *, const char *, const int);
-void greentea_send_kv(const char *, const char *, const int, const int);
+
+/**
+ * Encapsulate and send key-value message from DUT (device under test) to host
+ *
+ * @details This function uses underlying functions to write directly
+ *          to the serial port, (CONSOLE_TX). This allows KVs to be used
+ *          from within interrupt context.
```

Comment:
Is this correct? Sometimes we write to stdout, or somewhere else if the user provides their own implementation of the io abstraction layer.